[14.0][ADD] mrp_production_auto_validate

sebalix opened the pull request:
Auto-validate manufacturing orders when the "Pick components" transfer operation is validated.

This behavior is available only with manufacturing configured with 2 or 3 steps on the warehouse.

Ref cos-3796

sebalix committed:
[ADD] mrp_production_auto_validate

simahawk reviewed (approved):
LG

TDu reviewed (approved):
LG, small fix for the pre-commit to do.

grindtildeath commented on mrp_production_auto_validate/models/mrp_production.py:
```diff
@@ -0,0 +1,86 @@
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    auto_validate = fields.Boolean(
```
Shouldn't this field be related to the field on the BOM or at least computed? Relying on the onchange mechanism means it will be set to True only if the BOM is set on the Form view.

grindtildeath committed:
mrp_p_auto_validate: fix auto_validate flag

Relying on onchange makes this value set only if the BOM is set
manually on the Form view, what isn't the case when MO are created
by running procurements.

grindtildeath committed:
mrp_p_auto_validate: split qty on create

- split values for create in dedicated function
- post messages on MO with modified qty
- split MOs on create only on procurements

gurneyalex commented:
@sebalix can you check the CI status?

simahawk commented:
@sebalix any chance to fix this and get it merged? It's used in prod since month...

grindtildeath commented on mrp_production_auto_validate/tests/test_auto_validate.py:
```diff
@@ -0,0 +1,196 @@
+        self.assertEqual(len(created_mos), 1)
+        self.assertEqual(created_mos.product_qty, self.bom.product_qty)
+        self.assertFalse(any("split" in m.body for m in created_mos.message_ids))
+        self.assertFalse(any("raised" in m.body for m in created_mos.message_ids))
```
```suggestion
        self.assertFalse(any("increased" in m.body for m in created_mos.message_ids))
```

sebalix replied:
Thanks, tests are now green!

grindtildeath commented on mrp_production_auto_validate/tests/test_auto_validate.py:
```diff
@@ -0,0 +1,196 @@
+        )
+        self.assertEqual(len(created_mos), 1)
+        self.assertEqual(created_mos.product_qty, self.bom.product_qty)
+        self.assertTrue(any("raised" in m.body for m in created_mos.message_ids))
```
```suggestion
        self.assertTrue(any("increased" in m.body for m in created_mos.message_ids))
```

simahawk commented:
/ocabot merge nobump
